### PR TITLE
PLANET-3755 First version of social media block

### PR DESCRIPTION
**Pull request by kirdia**
Still needs refinement in editor.
Frontend renders as original block.

Instagram and Twitter embeds are render partially in editor.

Sample blocks:

Facebook
`
`
Instagram
`<!-- wp:planet4-blocks/social-media {"title":"title","description":"descriptionn","social_media_url":"https://www.instagram.com/p/B0o6tVFFH46","alignment_class":"aligncenter"} /-->`

Twitter
`<!-- wp:planet4-blocks/social-media {"title":"title","description":"descriptionn","social_media_url":"https://twitter.com/greenpeace_gr/status/1156929045638111232","alignment_class":"aligncenter"} /-->`

### CODE REVIEW

**Comment by kirdia** [comment]
Updated code to render embeds in the backend.
Disabled float on social-embed div, because it's breaking up preview in the editor.

Facebook embed
social_media_facebook


Instagram embed
social_media_instagram


Twitter embed
social_media_twitter

File: react-blocks/src/blocks/Socialmedia/Socialmedia.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            help={__('Optional', 'planet4-blocks-backend')}
+            value={this.props.description}
+            onChange={this.props.onDescriptionChange}
+          />
+        </div>
+
+        <div>
+          <RadioControl
+            label="Embed type"
+            help={__('Select oEmbed for the following types of social media<br>- Twitter: tweet, profile, list, collection, likes, moment<br>- Facebook: post, activity, photo, video, media, question, note<br>- Instagram: image', 'planet4-blocks-backend')}
```
Comment (comzeradd) [inline react-blocks/src/blocks/Socialmedia/Socialmedia.js]: The `<br>` html elements don't render properly here.

**Comment by kirdia** [comment]
Rebased and fixed html rendering issue

File: react-blocks/src/blocks/Socialmedia/Socialmedia.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    const {__} = wp.i18n;
+
+    const embed_type_help = __('Select oEmbed for the following types of social media<br>- Twitter: tweet, profile, list, collection, likes, moment<br>- Facebook: post, activity, photo, video, media, question, note<br>- Instagram: image', 'planet4-blocks-backend');
+
+
+    return (
+      <Fragment>
+        <div>
+          <TextControl
+            label="Title"
```
Comment (sagarsdeshmukh) [inline react-blocks/src/blocks/Socialmedia/Socialmedia.js]: Need to translate strings here.

File: react-blocks/src/blocks/Socialmedia/Socialmedia.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            placeholder="Enter title"
```
Comment (sagarsdeshmukh) [inline react-blocks/src/blocks/Socialmedia/Socialmedia.js]: same here.

File: react-blocks/src/blocks/Socialmedia/Socialmedia.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+            help={__('Optional', 'planet4-blocks-backend')}
+            value={this.props.title}
+            onChange={this.props.onTitleChange}
+          />
+        </div>
+
+
+        <div>
+          <TextareaControl
+            label="Description"
```
Comment (sagarsdeshmukh) [inline react-blocks/src/blocks/Socialmedia/Socialmedia.js]: Same here and rest of components `label` and  `placeholder` fields.

File: react-blocks/src/blocks/Socialmedia/SocialmediaBlock.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+          default: ''
+        },
+      },
+      // withSelect is a "Higher Order Component", it works as
+      // a Decorator, it will provide some basic API functionality
+      // through `select`.
+      edit: ({isSelected, attributes, setAttributes}) => {
+
+        // These methods are passed down to the
+        // Articles component, they update the corresponding attribute.
```
Comment (sagarsdeshmukh) [inline react-blocks/src/blocks/Socialmedia/SocialmediaBlock.js]: comment need to update.

**Comment by comzeradd** [comment]
@sagarsdeshmukh I see you already add the l10n function, so merging this

**Pull request merged**